packages/suite/src/hooks/suite/useFirmware.ts

```diff
@@ -22,12 +23,15 @@ export const useFirmware = () => {
     const firmware = useSelector(selectFirmware);
     const transport = useSelector(state => state.suite.transport);
     const modal = useSelector(state => state.modal);
+    const routeName = useSelector(selectRouteName);
     const { device } = useDevice();
 
     // Device in its state before installation is cached when installation begins.
     // Until then, access device as normal.
     const originalDevice = firmware.cachedDevice || device;
 
+    const shouldSwitchFirmwareType = routeName === 'firmware-type';
```

Comment:
I understand the benefit of not having to pass the prop anymore. But this sounds like a future problem. If someone moves the functionality to other page, or adds the functionality at other place, this won't work, because it's hardcoded to one specific route..

I'm fine with it, but at least, I think the `'firmware-type'` should be extracted as constant, which would also be used in `ForegroundAppModal.tsx`, and possibly other places. Idk.

Reply:
You're right, this is not nice. I refactored it in 7782d9366e8e0553c7c56ed37314d20fa7409996, but maybe I made it worse by having to pass it to the hook where it could be forgotten (yet I don't want to make it required because some components don't care about it). I could pass it to the functions as previously, but then I'd have to turn even `confirmOnDevice` to a function. I don't like either solution :) What do you think?